DOC-9626: 4.0 Migration work (guide, relnotes, ottoman etc.)

osfameron opened the pull request:
This remains on the SDK 3 *API Spec* migration guide.

Following main changes:

 * added wording about Node.js SDK 3.2 vs 4.0 and the SDK API 3 spec.
 * inlined a partial from `migration.adoc` to trial a cleanup of the
 unnecessary apologia paragraphs about semantic versioning.
 * included the api-version partial
   (see also https://github.com/couchbase/docs-sdk-common/pull/41
   as we have to cover the 2->4 case, so seems important to
   document SDK 2 for context.
 * removed commented Java code.
 * removed empty section.

NOTE: preview at https://osfameron.github.io/DOC-9626-NodeJS-4-migration/nodejs-sdk/current/project-docs/migrating-sdk-code-to-3.n.html

osfameron committed:
DOC-9626: 4.0 Migration Guide (wip)

This remains on the SDK 3 *API Spec* migration guide.

Following main changes:

 * added wording about Node.js SDK 3.2 vs 4.0 and the SDK API 3 spec.
 * inlined a partial from `migration.adoc` to trial a cleanup of the
unnecessary apologia paragraphs about semantic versioning.
 * included the api-version partial
(see also https://github.com/couchbase/docs-sdk-common/pull/41
as we have to cover the 2->4 case, so seems important to
document SDK 2 for context.
 * removed commented Java code.
 * removed empty section.

RichardSmedley commented:
Shouldn't this be going into the 4.0 branch?

osfameron commented:
> Shouldn't this be going into the 4.0 branch?

Yup, that would make sense 😊

maria-robobug commented on modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc:
```diff
@@ -11,11 +11,26 @@ include::partial$attributes.adoc[]
+include::{version-server}@sdk:shared:partial$api-version.adoc[tag=api-version]
+
+// include::{version-server}@sdk:shared:partial$migration.adoc[tag=intro]
+// tag::intro[]
```
Should this be uncommented?

RichardSmedley replied:
That's the semver section that Hakim mentioned removing.

osfameron replied:
I think no - it's just marking the location of the section that is *inlined* from that partial.
(Is non-obvious though..., could add a clearer comment maybe?)

osfameron replied:
that's right @RichardSmedley, I thought it was useful to keep the tag extents in case we choose to push that content back to the partial.

osfameron committed:
DOC-9626: Ottoman / 4.0 comments

Keep the page ref and location the same,
but reword as a general compatibility page.

osfameron committed:
DOC-9626: 4.0 compatibility

osfameron committed:
DOC-9626: 4.0 Migration Guide, Compat, Encryption

maria-robobug commented on modules/hello-world/pages/start-using-sdk.adoc:
```diff
@@ -6,12 +6,13 @@
+The Couchbase SDK API 3 (implemented by Node.js SDK 3._x_ and 4._x_) is a complete rewrite of the API, reducing the number of overloads to present a simplified surface area, and adding support for Couchbase Server features like xref:concept-docs:collections.adoc[Collections and Scopes] (available from Couchbase Server 7.0).
 
-The 3.x SDK also brings in link:https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises[promises], to reduce the complexity of asynchronous javascript in client applications, as well as extending the management APIs and bringing better debugging and logging options
+SDK 3 API also brings in link:https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises[promises], to reduce the complexity of asynchronous javascript in client applications, as well as extending the management APIs and bringing better debugging and logging options.
```
Unless I'm reading this wrong, should this be?
```suggestion
SDK API 3 also brings in link:https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises[promises], to reduce the complexity of asynchronous javascript in client applications, as well as extending the management APIs and bringing better debugging and logging options.
```

RichardSmedley committed:
Merge branch 'release/4.0' into DOC-9626-4.0-migration-guide

RichardSmedley committed:
Gardening

RichardSmedley committed:
Gardening

RichardSmedley committed:
Update modules/hello-world/pages/start-using-sdk.adoc

Thanks Maria

Co-authored-by: Maria Shodunke <maria-robobug@users.noreply.github.com>

RichardSmedley committed:
UK -->  US English

RichardSmedley reviewed (approved):
Thanks @osfameron 
LGTM  :+1:
